var/spack/packages/openssl/package.py

```diff
@@ -17,6 +17,10 @@ class Openssl(Package):
     parallel = False
 
     def install(self, spec, prefix):
+        # OpenSSL uses a variable APPS in its Makefile. If it happens to be set
+        # in the environment, then this will override what is set in the
+        # Makefile, leading to build errors.
+        del env['APPS']
```

Comment:
This will raise a `KeyError` if `APPS` is not set.  Can you use `env.pop('APPS', None)`?